### DISCUSSION TITLE
Get all projects for which the current user has push rights.

### DISCUSSION THREAD

**Discussion by ShawnRG** [goharbor/harbor#17912] [Q&A]
Hi,

I am looking for an easy way to get a list of projects that the current user has push rights to.

Thanks!

**Comment by AllForNothing** [comment]
@ShawnRG 

1. You can list all the projects for the current user.
`/api/v2.0/projects?page=1&page_size=200`,
the max page_size is 200,  so if the total projects number is more than 200,  then you have to call this for  total/200 + 1  times

2.  Then you filter the projects that its current_user_role_id property  is   `1 or 2 or 4`
```export const ProjectRoles = [
    { id: 1, value: 'MEMBER.PROJECT_ADMIN' },
    { id: 2, value: 'MEMBER.DEVELOPER' },
    { id: 3, value: 'MEMBER.GUEST' },
    { id: 4, value: 'MEMBER.PROJECT_MAINTAINER' },
    { id: 5, value: 'MEMBER.LIMITED_GUEST' },
];```

**Reply by ShawnRG** [reply]
I see!
Thanks for your answer, I will look into this.
